GitUI/CommandsDialogs/FormBrowse.cs

```diff
@@ -2161,13 +2161,29 @@ private void QuickFetch()
 
         public override bool ProcessHotkey(Keys keyData)
         {
-            return !IsDesignMode && HotkeysEnabled
-                && (base.ProcessHotkey(keyData) // generic handling of this form's hotkeys (upstream)
-                    || ((!GitExtensionsControl.IsTextEditKey(keyData) // downstream (without keys for quick search)
-                         || (keyData == (Keys.Control | Keys.A) && RevisionGridControl.ContainsFocus)) // route Ctrl+A away from RevisionGridControl
-                        && ((keyData != (Keys.Control | Keys.A) && RevisionGridControl.ProcessHotkey(keyData))
-                            || (CommitInfoTabControl.SelectedTab == DiffTabPage && revisionDiff.ProcessHotkey(keyData))
-                            || (CommitInfoTabControl.SelectedTab == TreeTabPage && fileTree.ProcessHotkey(keyData)))));
+            if (IsDesignMode || !HotkeysEnabled)
+            {
+                return false;
+            }
+
+            // generic handling of this form's hotkeys (upstream)
+            if (base.ProcessHotkey(keyData))
+            {
+                return true;
+            }
+
+            // downstream (without keys for quick search)
+            // but allow routing Ctrl+A away from RevisionGridControl in order to not select all revisions
+            if (GitExtensionsControl.IsTextEditKey(keyData)
+                && !(keyData == (Keys.Control | Keys.A) && RevisionGridControl.ContainsFocus))
+            {
+                return false;
+            }
+
+            // route to visible controls which have their own hotkeys
```

Comment:
This restructure makes this easier to read.
I also want to add an explanation.
(CommitInfo should have editable text for artificial commands, this could be changed then. You have added Side Panel hotkeys...)

```suggestion
            // route to visible controls which have their own hotkeys
            // For CommitInfo (as tab or panel), the text hotkeys are ignored
```

Reply:
I have added a comment regarding CommitInfo before `IsTextEditKey`.

> CommitInfo should have editable text for artificial commands, this could be changed then.

Yes, we must keep this in mind. When we add Hotkeys for CommitInfo, we must add it here, too.
Though I think "controls which have their own hotkeys" is clear enough.

> You have added Side Panel hotkeys...

... for a future follow-up PR. (The prepared branch does not work yet due to (shortcut use of) `PreviewKeyDown` in the ROT.)
Yes, I will add the routing to the ROT for completeness, although the existing hotkeys (rename, delete, search) have their own meaning for the RevisionGrid, which takes precedence.